### PR TITLE
Add South Australia's "March Public Holiday"

### PR DESCRIPTION
**Pull request by ghiculescu**
So aptly named because it's in March. http://www.safework.sa.gov.au/show_page.jsp?id=2483#.VQ9Mfmb8-8E

### CODE REVIEW

File: data/au.yaml [hunk 1]
```diff
@@ -234,4 +260,9 @@ tests: |
     assert_equal 'Royal Hobart Regatta', Date.civil(2012, 2, 13).holidays(:au_tas_south)[0][:name]
     assert_equal 'Royal Hobart Regatta', Date.civil(2014, 2, 10).holidays(:au_tas_south)[0][:name]
     assert_equal 'Royal Hobart Regatta', Date.civil(2015, 2, 9).holidays(:au_tas_south)[0][:name]
-    assert_equal 'Royal Hobart Regatta', Date.civil(2016, 2, 8).holidays(:au_tas_south)[0][:name]
\ No newline at end of file
+    assert_equal 'Royal Hobart Regatta', Date.civil(2016, 2, 8).holidays(:au_tas_south)[0][:name]
+
+    assert_equal "March Public Holiday", Date.civil(2014, 3, 10).holidays(:au_sa)[0][:name]
```
Comment (ptrimble) [inline data/au.yaml]: Would you mind adding in a test for dates before 2006 to show that it won't get picked up and to show exactly what the behavior will be?

**Comment by ptrimble** [comment]
@ghiculescu Thank you! Just one test-related comment.

**Comment by ghiculescu** [comment]
@ptrimble good thinking - done!

**Comment by ptrimble** [comment]
Sorry, this week is getting away from me!

I'm still a bit confused, don't we need a test that would show that nothing would be found for dates before the cutoff? Like so:

```ruby
assert_nil Date.civil(2001, 5, 16).holidays(:au_sa)[0][:name]
```

Or whatever the right date would be for both March and May examples.

**Comment by ghiculescu** [comment]
Ooops, I added another commit here that was meant for a separate PR: https://github.com/alexdunae/holidays/commit/8774b0b7897cb6992f2bf55f0418d7c1c4ed3659

That said I think it is fine to merge also?

**Comment by ptrimble** [comment]
@ghiculescu Yep! I don't have a problem with it. Leave it as-is.

Could you add in a test showing what happens if we pick a date before those cutoffs like I mentioned above? Or am I misunderstanding how this all works?

**Comment by ptrimble** [comment]
FYI, while performing some refactoring I've stumbled across the kind of test I was thinking about:

```ruby
assert_equal [], Holidays.on(Date.civil(2009,1,6), region), "Heilige Drei Könige is not a holiday in #{region}"
```

I just want something to show that the cutoff is working.

**Comment by ghiculescu** [comment]
Ooops, sorry for the slackness. Pushed now.

**Comment by ptrimble** [comment]
Beeeeyoootiful. Thanks for dealing with my OCD surrounding tests! :+1:

**Pull request merged**